commit 6e6d194423aa51c521e1201104687dbd366ee50d
Author: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
Date:   Wed Jan 4 06:30:16 2017 +0100

    usb: gadget: composite: Fix function used to free memory
    
    commit 990758c53eafe5a220a780ed12e7b4d51b3df032 upstream.
    
    'cdev->os_desc_req' has been allocated with 'usb_ep_alloc_request()' so
    'usb_ep_free_request()' should be used to free it.
    
    Signed-off-by: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
    Signed-off-by: Felipe Balbi <felipe.balbi@linux.intel.com>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index c3c5b87..baa7cdc 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -2147,7 +2147,7 @@ int composite_os_desc_req_prepare(struct usb_composite_dev *cdev,
 	cdev->os_desc_req->buf = kmalloc(4096, GFP_KERNEL);
 	if (!cdev->os_desc_req->buf) {
 		ret = -ENOMEM;
-		kfree(cdev->os_desc_req);
+		usb_ep_free_request(ep0, cdev->os_desc_req);
 		goto end;
 	}
 	cdev->os_desc_req->context = cdev;
